PlayList.java

```diff
@@ -71,34 +83,60 @@ public int indexOf(String title) {
      *  is full, does nothing and returns false. Otherwise, inserts the track and
      *  returns true. */
     public boolean add(int i, Track track) {
-        //// replace the following statement with your code
-        return false;
+        if (i < 0 || i > maxSize || size == maxSize) return false;
+        if (this.size == 0) {
+            this.add(track); 
+            size++;
+            return true;
+        } else {
+            for (int trackNum = size - 1; trackNum >= i; trackNum--) {
+                tracks[trackNum + 1] = tracks[trackNum];
+            }
+            tracks[i] = track;
+            size++;
+            return true;
+        }
     }
      
     /** Removes the track in the given index from this list.
      *  If the list is empty, or the given index is negative or too big for this list, 
-     *  does nothing and returns -1. */
+     *  does nothing. */
     public void remove(int i) {
-        //// replace this comment with your code
+        if (size == 0 || i < 0 || i >= size) return;
+        for (int trackNum = i; trackNum < size - 1; trackNum++) {
+            tracks[trackNum] = tracks[trackNum + 1];
+        }
+        size--;;
     }
 
     /** Removes the first track that has the given title from this list.
-     *  If such a track is not found, or the list is empty, or the given index
-     *  is negative or too big for this list, does nothing. */
+     *  If such a track is not found, or the list is empty. */
     public void remove(String title) {
-        //// replace this comment with your code
+        if (size == 0) return;
+        for (int trackNum = 0; trackNum < size; trackNum++) {
+            String currentTrackTitle = tracks[trackNum].getTitle().toLowerCase();
+            String inputTitle = title.toLowerCase();
+            if (currentTrackTitle.equals(inputTitle)) {
+                this.remove(trackNum);
+                return;
+            }
+        }
```

Comment:
You already implemented it, let's reuse it :)
```suggestion
        int index = indexOf(title);
        if (index >= 0 && index < this.size) {
             remove(index);
        }
```